Core/src/org/sleuthkit/autopsy/datamodel/SCOAndTranslationTask.java

```diff
@@ -0,0 +1,144 @@
+import org.sleuthkit.autopsy.events.AutopsyEvent;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.ContentTag;
+
+
+/**
+ * Completes the tasks needed to populate the Score, Comment, Occurrences and Translation
+ * columns in the background so that the UI is not blocked while waiting for responses from the database or 
+ * translation service. Once these events are done, it fires a PropertyChangeEvent
+ * to let the AbstractAbstractFileNode know it's time to update!
+ */
+class SCOAndTranslationTask implements Runnable {
+    
+    private final WeakReference<AbstractFile> weakContentRef;
+    private final PropertyChangeListener listener;
+    
+    public SCOAndTranslationTask(WeakReference<AbstractFile> weakContentRef, PropertyChangeListener listener) {
+        this.weakContentRef = weakContentRef;
+        this.listener = listener;
+    }
+
+    @Override
+    public void run() {
+        try {
+            AbstractFile content = weakContentRef.get();
+            
+            //Long DB queries
+            List<ContentTag> tags = FileNodeUtil.getContentTagsFromDatabase(content);
+            CorrelationAttributeInstance attribute = 
+                    FileNodeUtil.getCorrelationAttributeInstance(content);
+
+            Pair<DataResultViewerTable.Score, String> scoreAndDescription = 
+                    FileNodeUtil.getScorePropertyAndDescription(content, tags);
+            DataResultViewerTable.HasCommentStatus comment = 
+                    FileNodeUtil.getCommentProperty(tags, attribute);
+            Pair<Long, String> countAndDescription = 
+                    FileNodeUtil.getCountPropertyAndDescription(attribute);
+
+            //Load the results from the SCO column operations into a wrapper object to be passed
+            //back to the listener so that the node can internally update it's propertySheet.
+            SCOResults results = new SCOResults(
+                    scoreAndDescription.getLeft(),
+                    scoreAndDescription.getRight(),
+                    comment,
+                    countAndDescription.getLeft(),
+                    countAndDescription.getRight()
+            );
+
+            listener.propertyChange(new PropertyChangeEvent(
+                AutopsyEvent.SourceType.LOCAL.toString(),
+                AbstractAbstractFileNode.NodeSpecificEvents.DABABASE_CONTENT_AVAILABLE.toString(),
+                null,
+                results));
+
+            //Once we've got the SCO columns, then lets fire the translation result.
+            //Updating of this column is significantly lower priority than 
+            //getting results to the SCO columns!
+            listener.propertyChange(new PropertyChangeEvent(
+                AutopsyEvent.SourceType.LOCAL.toString(),
+                AbstractAbstractFileNode.NodeSpecificEvents.TRANSLATION_AVAILABLE.toString(),
+                null,
+                FileNodeUtil.getTranslatedFileName(content)));
```

Comment:
Is it possible that we are dispatching a task to the EDT that will often do nothing because there is no translation provider or the translation has already been done? I don't know that it really merits much attention to optimization, but we could make the firing of the property change event conditional on the result of getTranslatedFileName.